quickwit-search/src/root.rs

```diff
@@ -113,6 +120,35 @@ async fn execute_search(
     Ok(result_per_node_addr)
 }
 
+async fn execute_fetch_docs(
+    search_request: &SearchRequest,
+    partial_hits_map: &HashMap<String, Vec<PartialHit>>,
+    doc_fetch_jobs: &[(SearchServiceClient, Vec<Job>)],
+) -> anyhow::Result<Vec<Result<FetchDocsResult, SearchError>>> {
+    // Perform the fetch docs phase.
+    let mut fetch_docs_handles: Vec<JoinHandle<Result<FetchDocsResult, SearchError>>> = Vec::new();
```

Comment:
I know this is not related to this PR, but we don't need to allocate a vector here: `futures::future::try_join_all` can resolve an iterator of futures.